ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/CommandTopicBackupTest.java

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.server;
+
+import com.google.common.base.Ticker;
+import io.confluent.ksql.rest.entity.CommandId;
+import io.confluent.ksql.rest.server.computation.Command;
+import io.confluent.ksql.util.KsqlException;
+import io.confluent.ksql.util.Pair;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CommandTopicBackupTest {
+  private static final String COMMAND_TOPIC_NAME = "command_topic";
+
+  private Pair<CommandId, Command> command1 = newStreamRecord("stream1");
+  private Pair<CommandId, Command> command2 = newStreamRecord("stream2");
+  private Pair<CommandId, Command> command3 = newStreamRecord("stream3");
+
+  @Mock
+  private Ticker ticker;
+
+  @Rule
+  public TemporaryFolder backupLocation = new TemporaryFolder();
+
+  private CommandTopicBackup commandTopicBackup;
+
+  @Before
+  public void setup() {
+    commandTopicBackup = new CommandTopicBackup(
```

Comment:
can we add a test where the backup file is corrupted and we can't properly read it?